Plugins Dashboard: Restore plugins/sites side by side view

xavier-lc opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Part of DOTMSD-876

## Proposed Changes

* Show the plugins list and the sites where the plugin is installed on side by side. 

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* The current Multi-Site Dashboard version requires an extra click to see where a plugin is installed. This adds friction and makes bulk evaluation/management harder.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Open the live preview
* Go to `/v2/plugins`
* Check that you can click on plugins to see what sites they are installed on

|Before|After|
|---|---|
|<img width="3842" height="1917" alt="image" src="https://github.com/user-attachments/assets/1bf242be-3686-4987-805f-3c5d38a00ec2" />|<img width="3842" height="1917" alt="image" src="https://github.com/user-attachments/assets/11079b41-7ee4-4b94-9d24-c16470ea8881" />|

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Commit:
Get the 2 columns in place

Commit:
Handle routing

Commit:
Match the plugin order on both sides

Commit:
Add header

Commit:
Adjust Grid

Commit:
Styling

Commit:
Add dataviews like hover

Commit:
Fix html

Commit:
Styling

Commit:
Styling

Commit:
Add title prop for long names

Commit:
Rm site icon

Commit:
Shorten texts

Commit:
Rm unnecessary nodes

Commit:
Styling

Commit:
Adjust spacing

Commit:
Adjust icons

Commit:
Adjust spacing

Commit:
Make plugins scrollable

Commit:
Rm leftover code

Commit:
Rm now unused `Plugin` component

Commit:
Rm unused fields

Commit:
Rm unused actions

Commit:
Rm unused views

Commit:
Rm unused type

Commit:
Move style to the right file

Commit:
Fix eslint

rcrdortiz commented:
I can see a vertical scrollbar for the whole screen, even though there is no apparent reason for scrolling vertically.

<img width="1529" height="1287" alt="Screenshot 2025-12-11 at 16 40 05" src="https://github.com/user-attachments/assets/ad7cab53-d598-47db-acc8-de60b9726d3f" />

rcrdortiz commented:
The site icon is missing now on the site rows.

Commit:
Fix eslint

rcrdortiz commented on client/dashboard/plugins/manage/index.tsx:
```diff
@@ -112,31 +153,98 @@ export default function PluginsList() {
+							searchClassName="plugin-switcher-search"
+							initialView={ view }
+							items={ pluginsWithIcon }
+							getItemUrl={ ( item ) => `/plugins/manage/${ item.slug }` }
```
Could we use `route.to` or a similar approach for building the item URL?

xavier-lc replied:
Addressed on https://github.com/Automattic/wp-calypso/pull/107633/commits/90e45c15b014c5ac7587d1e637212eb70f953b19

Commit:
Make scrolling conditional

lucasmendes-design commented:
It's looking great, folks. Some UI direction so we can keep everything consistent. 

### Tabs

Below is the screenshot of Manage Plugins in Figma, the PR, and Logs pages. 

<img width="1916" height="1824" alt="Group 2147226042" src="https://github.com/user-attachments/assets/1b712fdd-c0e3-4731-bf3f-1c8d81580933" />

### Plugin Icon

Some plugins don't have a plugin icon. Also, the plugin icon is too big. Can we reduce it? 

<img width="680" height="370" alt="image" src="https://github.com/user-attachments/assets/97c7a31e-0133-447e-b238-f1357fd14370" />
<img width="268" height="79" alt="image" src="https://github.com/user-attachments/assets/36a12376-ff85-4bb9-acaf-7bffc51f2257" />


### Select state

It's missing the select state in the Plugins list. 

### Spacing

We can add more spacing here. 

<img width="749" height="361" alt="image" src="https://github.com/user-attachments/assets/f67179cb-4d12-4d9c-be5c-209565399f44" />

Commit:
Prevent scroll to top on navigation

Commit:
Prevent plugin list from growing too large

xavier-lc commented:
> Are you comfortable using Figma's dev mode? Let me know the best way I can share with you the design specs :)

@lucasmendes-design sure, the figma link would be great.

Commit:
Build link out of route

Commit:
Fix tabs style

Commit:
Adjust icon size

Commit:
Selected style

Commit:
Merge branch 'trunk' into DOTMSD-876/restore-side-by-side-view-for-plugin-management

Commit:
Fix selected state

StevenDufresne commented:
Can we improve the mobile experience?

<img width="300" height="963" alt="Screenshot 2025-12-12 at 6 43 57 AM" src="https://github.com/user-attachments/assets/544ade20-ee08-4de8-aae0-21e55ddec016" />

lucasmendes-design commented:
> sure, the figma link would be great.

What is the placeholder we normally use? Is the one below? 
<img width="51" height="60" alt="image" src="https://github.com/user-attachments/assets/6e7810d6-f2e6-4b81-a840-112d6f2d080f" />

Can we change it?

rcrdortiz commented:
> Can we improve the mobile experience?
> 
> <img alt="Screenshot 2025-12-12 at 6 43 57 AM" width="300" height="963" src="https://private-user-images.githubusercontent.com/1657336/525610780-544ade20-ee08-4de8-aae0-21e55ddec016.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU1Mjc2NjAsIm5iZiI6MTc2NTUyNzM2MCwicGF0aCI6Ii8xNjU3MzM2LzUyNTYxMDc4MC01NDRhZGUyMC1lZTA4LTRkZTgtYWFlMC0yMWU1NWRkZWMwMTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTIxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEyMTJUMDgxNjAwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2MzMTE4MzA0Y2U3OTMwNzIzOWZjYTEwYzI0M2YzYWE2NmJkMGJjZmU4YjAzYzUzMzg1YjY2NDU2ZGU3ZDRiOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.s_6funVsJxmPfOR8Rq_FWrPEAk7_u4qNb8pVgjdkeVA">

@lucasmendes-design flagging that the mobile experience is currently unusable in this PR. We’ll need design input on the expected mobile behavior before merging. Any guidance?

fushar commented on client/dashboard/app/router/plugins.tsx:
```diff
@@ -59,9 +59,9 @@ export const pluginsManageRoute = createRoute( {
 } );
 
-export const pluginsManageIndexRoute = createRoute( {
+export const pluginRouteWithId = createRoute( {
```
We can actually simplify these two routes using the original structure like the following:

```ts
export const pluginsManageIndexRoute = createRoute( {
	getParentRoute: () => pluginsManageRoute,
	path: '/',
} ).lazy( () =>
	import( '../../plugins/manage' ).then( ( d ) =>
		createLazyRoute( 'plugins-manage' )( {
			component: d.default,
		} )
	)
);

export const pluginRoute = createRoute( {
	getParentRoute: () => pluginsManageIndexRoute,
	path: '$pluginId',
} );
```

xavier-lc replied:
Good catch, thanks, addressed on https://github.com/Automattic/wp-calypso/pull/107633/commits/7b6032d6aabd862ad66e52253ca4f182e8b1e4d6

fushar commented on client/dashboard/plugins/manage/index.tsx:
```diff
@@ -5,51 +5,85 @@ import {
+const view: View = {
+	type: 'list',
+	page: 1,
+	perPage: 100,
```
I assume infinite scrolling will be added later? Or we increase this value somehow.

xavier-lc replied:
@fushar created a followup issue: https://linear.app/a8c/issue/DOTMSD-956/handle-infinite-scrolling-on-the-plugins-list

lucasmendes-design commented:
> Aside from this, and the mobile view issue (which I think we can address in a follow-up once we get input from @lucasmendes-design, so we don’t block this PR), the changes look good to go 👍

+1. On my way, by the end of the day I'll p2 it :)

lucasmendes-design commented:
Can we add the search field the same as the "Installed on X" tab? 

<img width="915" height="452" alt="image" src="https://github.com/user-attachments/assets/1104e6fa-a5ed-402e-917b-c176973c4e91" />

<img width="920" height="300" alt="image" src="https://github.com/user-attachments/assets/6a210636-7cf1-4c28-b492-8904bd445e85" />

lucasmendes-design commented:
What does the copy "Plugin up to date" mean? For example, I have a plugin that has 1 update available and the copy still shows "Plugin up to date".  

<img width="1163" height="460" alt="image" src="https://github.com/user-attachments/assets/c488b38f-3b45-482a-964f-dd4fe75ea443" />

Commit:
Address PR feedback

lucasmendes-design commented:
Some suggestions in terms of layout: 

Remove the primary action "WP-Admin". This table doesn't have a primary action like that. The main actions are the Active and Auto-update toggles. This way, we get more space for the left panel so we can add the filter Update available. 

Left panel: 392px 

<img width="1045" height="349" alt="image" src="https://github.com/user-attachments/assets/7c905aa4-3fd3-41bb-ab1b-6498d445e059" />

lucasmendes-design commented:
All the toggles have a light gray text "On" and "Off". Which conveys that the toggle is disable. 

<img width="826" height="429" alt="image" src="https://github.com/user-attachments/assets/6783a404-d1fb-41ce-ae93-cee0a2285e0a" />

The correct form is the one below. 

<img width="410" height="280" alt="image" src="https://github.com/user-attachments/assets/eff3ddf7-6642-4697-bdf1-f3e678698105" />

**Question:** When a plugin is updated automatically, will users receive the update automatically? 

- If yes, I'd make the toggle On + Disable. 
- If no, Id make the toggle Off + Disable (this is the approach we're using it)

Commit:
Prevent scrolling on the dataview

Commit:
Enable search on available tab

Commit:
Rm wp-admin action

Commit:
Adjust left col size

Commit:
Change copy

Commit:
Fix toggle text color

Commit:
Adjust search size

Commit:
Adjust font size

xavier-lc commented:
> Could we keep the DataView fixed and scope scrolling only to the navigation pane?

@rcrdortiz addressed on https://github.com/Automattic/wp-calypso/pull/107633/changes/e6bb70b28f26ce1187ca198d4e70fa7a51dae598

> Can we add the search field the same as the "Installed on X" tab?

@lucasmendes-design addressed on https://github.com/Automattic/wp-calypso/pull/107633/changes/d24293860d67bd25f09d9fd5466b6a9c13dacaad

> What does the copy "Plugin up to date" mean? For example, I have a plugin that has 1 update available and the copy still shows "Plugin up to date".

@lucasmendes-design it means there are no updates available, not sure what's going on your case 😅

<img width="2124" height="1522" alt="image" src="https://github.com/user-attachments/assets/f9aa20ec-0a1c-44ec-b4c5-055995f9c7ac" />
<img width="2124" height="1522" alt="image" src="https://github.com/user-attachments/assets/5d766f7e-386f-4379-945a-e209096fe208" />

> Remove the primary action "WP-Admin". This table doesn't have a primary action like that. The main actions are the Active and Auto-update toggles. This way, we get more space for the left panel so we can add the filter Update available.

Addressed on https://github.com/Automattic/wp-calypso/pull/107633/changes/ea5c0fda0cdf19484d3ae57d7f89e44b95bcc5eb

> Left panel: 392px

Addressed on https://github.com/Automattic/wp-calypso/pull/107633/changes/9dc283566f350e5f49cb81f32435d01cae0b2ee1

> The column name is already Update. So, I think we can change from Updates Auto-managed to Auto-managed

Addressed on https://github.com/Automattic/wp-calypso/pull/107633/changes/70f95afbec5ea9c2d61cea340a1a2fd267fbc13b

> All the toggles have a light gray text "On" and "Off". Which conveys that the toggle is disable.

Addressed on https://github.com/Automattic/wp-calypso/pull/107633/changes/c9a1c4f482b50b8c977fb108a45a42ee03d4cd39

> The search input field is fixed and there is no need to make it full-width. You can keep the same size as Sites, Plugins, etc.

Addressed on https://github.com/Automattic/wp-calypso/pull/107633/changes/322f5a0ef7cb20fa03e4cbf7f90a1dc8392cb51d

> Also, the tab font sizes are a bit too big. Let's use what we have in Logs. ✌🏽

Addressed on https://github.com/Automattic/wp-calypso/pull/107633/changes/fe476bd2253dac2167ae2f1c7b17335dce4a814f

rcrdortiz commented:
> > To keep this PR moving, could we focus it on the changes already scoped here and capture the broader design review items as follow-ups?
> 
> Of course, sorry for the spammy comments 😆 I thought the idea was to review the details already.
> 
> When you say "follow-ups", do you mean before we launch, correct? Happy to help.

No worries at all! The comments are helpful 🙂

By follow-ups I mean separate issues that we can prioritize intentionally, not necessarily before launch.
For this PR, I’d suggest we stick to the scoped changes (transition from 2 Dataview to side-by-side display) so we can land it safely, and then decide which design refinements make sense to tackle next and on what timeline.

fushar commented on client/dashboard/app/router/plugins.tsx:
```diff
@@ -71,26 +71,9 @@ export const pluginsManageIndexRoute = createRoute( {
-	getParentRoute: () => pluginsManageRoute,
+	getParentRoute: () => pluginsManageIndexRoute,
 	path: '$pluginId',
-	loader: async () => {
```
Ah, should we keep this loader? I think the prefetches are useful.

xavier-lc replied:
You're right, addressed on https://github.com/Automattic/wp-calypso/pull/107633/commits/0ac6ef7485f8effc85a2f5d5303b48ae0007ce4f

Commit:
Bring back prefetch

Commit:
Fix lint
